src/components/ScenarioParameters/components/ScenarioParametersInputs/GenericTable.js

```diff
@@ -209,76 +233,107 @@ export const GenericTable = ({
-          )
-        );
-      const csvRows = AgGridUtils.toCSV(rowsDict, columns);
-      const agGridData = _generateGridDataFromCSV(csvRows, parameterData);
-      if (agGridData.error) {
-        setClientFileDescriptor({
-          tableDataStatus: TABLE_DATA_STATUS.ERROR,
-          errors: agGridData.error,
-        });
-      } else
-        setClientFileDescriptor({
-          name: fileName,
-          file: null,
-          agGridRows: agGridData.rows,
-          errors: null,
-          status: UPLOAD_FILE_STATUS_KEY.READY_TO_UPLOAD,
-          tableDataStatus: TABLE_DATA_STATUS.READY,
-          uploadPreprocess: { content: _uploadPreprocess },
-        });
+      ({ data } = await Api.Datasets.twingraphQuery(organizationId, sourceDatasetId, { query }));
     } catch (error) {
-      const errorComment = error?.response?.status
-        ? t(
-            'genericcomponent.table.labels.notTwingraphDatasetError',
-            'Only twingraph datasets can be used to fetch table data dynamically'
+      setPlaceholderTitle(
+        error?.response?.statusText ?? t('genericcomponent.table.labels.queryFailedError', 'Query failed')
+      );
+      setPlaceholderBody(
+        error?.response?.data?.detail ??
+          t(
+            'genericcomponent.table.labels.unknownQueryError',
+            'Something went wrong when querying the table data. Please contact your administrator if the ' +
+              'problem persists.'
           )
-        : '';
-      setApplicationErrorMessage(error, errorComment);
+      );
+
+      _setClientFileDescriptorToNull();
+      return;
+    }
+
+    if (data.length === 0) {
+      setPlaceholderTitle(t('genericcomponent.table.labels.noDataPlaceholderTitle', 'No data'));
+      setPlaceholderBody(
+        t(
+          'genericcomponent.table.labels.wrongQueryError',
+          'Returned result is empty, there is probably an error in your query configuration. ' +
+            'Please, check your solution'
+        )
+      );
+      return;
+    }
+
+    const rowsDict = data.map((row) => row[resultKey]);
+    if (rowsDict.includes(undefined)) {
+      setPlaceholderTitle(t('genericcomponent.table.labels.configErrorPlaceholderTitle', 'Configuration error'));
+      setPlaceholderBody(
+        t(
+          'genericcomponent.table.labels.wrongResultKeyError',
+          `Returned result doesn't have ${resultKey} property.
+            Probably there is an error in resultKey configuration, please, check your solution`,
+          { resultKey }
+        )
+      );
+      return;
+    }
+
+    if (findDatasetById(sourceDatasetId)?.ingestionStatus == null) {
+      setPlaceholderTitle(t('genericcomponent.table.labels.noTwingraph', 'No Twingraph'));
+      setPlaceholderBody(
+        t(
+          'genericcomponent.table.labels.notTwingraphDatasetError',
+          'Only twingraph datasets can be used to fetch table data dynamically'
+        )
+      );
+      return;
+    }
```

Comment:
I think it's better to put this error before the API call because the query will fail anyway so we don't need to send it if the dataset isn't twingraph